app/src/main/java/com/example/teamyuml/decentworkmobile/Login.java

```diff
@@ -1,4 +1,136 @@
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        this.requestWindowFeature(Window.FEATURE_NO_TITLE);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login);
+        loginInputEmail = (EditText) findViewById(R.id.login_input_email);
+        loginInputPassword = (EditText) findViewById(R.id.login_input_password);
+        btnlogin = (Button) findViewById(R.id.btn_login);
+        btnLinkSignup = (Button) findViewById(R.id.btn_link_signup);
+        // Progress dialog
+        progressDialog = new ProgressDialog(this);
+        progressDialog.setCancelable(false);
+
+        btnlogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                loginUser(loginInputEmail.getText().toString(),
+                        loginInputPassword.getText().toString());
+            }
+        });
+
+        btnLinkSignup.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Intent i = new Intent(getApplicationContext(), Registration.class);
+                startActivity(i);
+            }
+        });
+    }
+
+    private void loginUser( final String email, final String password) {
+        // Tag used to cancel the request
+        String cancel_req_tag = "login";
+        progressDialog.setMessage("Logging you in...");
+        showDialog();
+        StringRequest strReq = new StringRequest(Request.Method.POST,
+                URL_FOR_LOGIN, new Response.Listener<String>() {
+
+            @Override
+            public void onResponse(String response) {
+                Log.d(TAG, "Register Response: " + response.toString());
+                hideDialog();
+                try {
+                    JSONObject jObj = new JSONObject(response);
+                    boolean error = jObj.getBoolean("error");
+
+                    if (!error) {
+                        String user = jObj.getJSONObject("user").getString("name");
+                        // Launch User activity
+                        Intent intent = new Intent(
+                                Login.this,
+                                SplashScreen.class);
+                        intent.putExtra("username", user);
+                        startActivity(intent);
+                        finish();
+                    }
+                    else {
+
+                        String errorMsg = jObj.getString("error_msg");
+                        Toast.makeText(getApplicationContext(),
+                                errorMsg, Toast.LENGTH_LONG).show();
+                    }
+                }
+                catch (JSONException e) {
+                    e.printStackTrace();
+                }
+            }
+        }, new Response.ErrorListener() {
+            @Override
+            public void onErrorResponse(VolleyError error) {
```

Comment:
I would rather check for HTTP status codes from response and then adjust toast message to them + idk if we want toast rather than text field or something but here @KamilSupera should answer.

Reply:
U can get status codes with error.networkResponse.statusCode